### PR TITLE
feat: use new pod identity based tags

### PR DESCRIPTION
**Pull request by rschalo**
Fixes #N/A 

**Description**
Update to use EKS controller tagging standards

**How was this change tested?**

**Does this change impact docs?**
- [ ] Yes, PR includes docs updates 
- [ ] Yes, issue opened: # 
- [x] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by ellistarn** [comment]
Do we consider this breaking or an implementation detail?

**Comment by rschalo** [comment]
> Do we consider this breaking or an implementation detail?

I think this is an implementation detail.

I tested this by creating a deployment with one replica with the old annotation, waiting for a node to come online, upgrading karpenter to an image with the new annotation, scaling out the deployment to two replicas after the upgrade, and then waiting for the second node to come up. 

Both nodes continued to operate just fine and later when a `SportInterruption` disruption happened to the node with the `karpenter.sh/managed-by` annotation, the node rotated out as expected.

File: pkg/apis/v1beta1/ec2nodeclass.go [hunk 1]
```diff
@@ -356,7 +355,7 @@ func (in *EC2NodeClass) InstanceProfileRole() string {
 func (in *EC2NodeClass) InstanceProfileTags(clusterName string) map[string]string {
 	return lo.Assign(in.Spec.Tags, map[string]string{
 		fmt.Sprintf("kubernetes.io/cluster/%s", clusterName): "owned",
-		corev1beta1.ManagedByAnnotationKey:                   clusterName,
+		EksClusterNameAnnotationKey:                          clusterName,
```
Comment (njtran) [inline pkg/apis/v1beta1/ec2nodeclass.go:358]: nit: follow the Go semantic of capitalizing the acronym
```suggestion
		EKSClusterNameAnnotationKey:                          clusterName,
```

File: pkg/apis/v1beta1/labels.go [hunk 2]
```diff
@@ -69,10 +69,13 @@ var (
 		// https://docs.aws.amazon.com/eks/latest/APIReference/API_CreateCluster.html
 		regexp.MustCompile(`^kubernetes\.io/cluster/[0-9A-Za-z][A-Za-z0-9\-_]*$`),
 		regexp.MustCompile(fmt.Sprintf("^%s$", regexp.QuoteMeta(v1beta1.NodePoolLabelKey))),
-		regexp.MustCompile(fmt.Sprintf("^%s$", regexp.QuoteMeta(v1beta1.ManagedByAnnotationKey))),
+		regexp.MustCompile(fmt.Sprintf("^%s$", regexp.QuoteMeta(EksClusterNameAnnotationKey))),
 		regexp.MustCompile(fmt.Sprintf("^%s$", regexp.QuoteMeta(LabelNodeClass))),
 		regexp.MustCompile(fmt.Sprintf("^%s$", regexp.QuoteMeta(TagNodeClaim))),
 	}
+
+	EksClusterNameAnnotationKey = "eks:eks-cluster-name"
```
Comment (njtran) [inline pkg/apis/v1beta1/labels.go:77]: Same here
```suggestion
	EKSClusterNameAnnotationKey = "eks:eks-cluster-name"
```

File: pkg/providers/launchtemplate/launchtemplate.go [hunk 3]
```diff
@@ -271,7 +271,11 @@ func (p *DefaultProvider) createLaunchTemplate(ctx context.Context, options *ami
 		TagSpecifications: []*ec2.TagSpecification{
 			{
 				ResourceType: aws.String(ec2.ResourceTypeLaunchTemplate),
-				Tags:         utils.MergeTags(options.Tags, map[string]string{v1beta1.TagManagedLaunchTemplate: options.ClusterName, v1beta1.LabelNodeClass: options.NodeClassName}),
+				Tags: utils.MergeTags(options.Tags, map[string]string{
+					v1beta1.EksClusterNameAnnotationKey: options.ClusterName,
```
Comment (njtran) [inline pkg/providers/launchtemplate/launchtemplate.go:275]: So this adds the eks cluster annotation tag on now, is this right?

File: pkg/apis/v1beta1/labels.go (hunk 2 above)
Comment (njtran) [reply pkg/apis/v1beta1/labels.go:77]: This isn't viewed as a system tag, right? Are we sure that we'll always be able to do this?

File: test/hack/resource/pkg/resourcetypes/resourcetypes.go [hunk 4]
```diff
@@ -20,7 +20,7 @@ import (
 )
 
 const (
-	karpenterClusterNameTag    = "karpenter.sh/managed-by"
+	karpenterClusterNameTag    = "eks:eks-cluster-name"
```
Comment (njtran) [inline test/hack/resource/pkg/resourcetypes/resourcetypes.go:23]: Can't rely on the v1beta1 label?

File: test/suites/integration/tags_test.go [hunk 5]
```diff
@@ -86,7 +86,7 @@ var _ = Describe("Tags", func() {
 	})
 
 	Context("Tagging Controller", func() {
-		It("should tag with karpenter.sh/nodeclaim and Name tag", func() {
+		It("should tag with karpenter.sh/nodeclaim, eks:cluster-name, and Name tag", func() {
```
Comment (njtran) [inline test/suites/integration/tags_test.go:89]:
```suggestion
		It("should tag with karpenter.sh/nodeclaim, eks:eks-cluster-name, and Name tag", func() {
```

File: test/suites/integration/validation_test.go [hunk 6]
```diff
@@ -194,7 +194,7 @@ var _ = Describe("Validation", func() {
 			nodeClass.Spec.Tags = map[string]string{"karpenter.sh/nodepool": "custom-value"}
 			Expect(env.Client.Create(env.Context, nodeClass)).ToNot(Succeed())
 
-			nodeClass.Spec.Tags = map[string]string{"karpenter.sh/managed-by": env.ClusterName}
+			nodeClass.Spec.Tags = map[string]string{"eks:eks-cluster-name": env.ClusterName}
```
Comment (njtran) [inline test/suites/integration/validation_test.go:197]: Seems odd that we would define the const but not use it here. Any reason why we're not?

File: pkg/apis/v1beta1/ec2nodeclass.go [hunk 7]
```diff
@@ -80,7 +79,7 @@ type EC2NodeClassSpec struct {
 	// +kubebuilder:validation:XValidation:message="empty tag keys aren't supported",rule="self.all(k, k != '')"
 	// +kubebuilder:validation:XValidation:message="tag contains a restricted tag matching kubernetes.io/cluster/",rule="self.all(k, !k.startsWith('kubernetes.io/cluster') )"
 	// +kubebuilder:validation:XValidation:message="tag contains a restricted tag matching karpenter.sh/nodepool",rule="self.all(k, k != 'karpenter.sh/nodepool')"
-	// +kubebuilder:validation:XValidation:message="tag contains a restricted tag matching karpenter.sh/managed-by",rule="self.all(k, k !='karpenter.sh/managed-by')"
+	// +kubebuilder:validation:XValidation:message="tag contains a restricted tag matching eks:eks-cluster-name",rule="self.all(k, k !='eks:eks-cluster-name')"
```
Comment (ellistarn) [inline pkg/apis/v1beta1/ec2nodeclass.go:82]: Curious where this standard comes from. This stutters the word "eks".

File: test/suites/integration/validation_test.go (hunk 6 above)
Comment (rschalo) [reply test/suites/integration/validation_test.go:197]: No reason, updated.

File: pkg/providers/launchtemplate/launchtemplate.go (hunk 3 above)
Comment (rschalo) [reply pkg/providers/launchtemplate/launchtemplate.go]: Correct, according to @mikestef9, it should be added to all resources Karpenter creates.

**Pull request closed without merging**